apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.ISessionPool.cs

```diff
@@ -30,6 +30,7 @@ internal interface ISessionPool
             IClock Clock { get; }
             Task<PooledSession> WithFreshTransactionOrNewAsync(PooledSession session, TransactionOptions transactionOptions, CancellationToken cancellationToken);
             void Release(PooledSession session, ByteString transactionToRollback, bool deleteSession);
+            void Detach(PooledSession session);
```

Comment:
None of the implementations use the session - that sounds like it's either a bug (probably in TargetedSessionPool) or that we can remove the parameter.

Reply:
Ping on this one.

Reply:
It's not a bug, TargetetSessionPool does not keep track of PooledSession instances themselves, just through counters.

None of this is public so we can definetely remove the parameter. But then I think I'd like to call this method something like DetachActiveSession. Does that sound good?

Reply:
Either that, or we could document that the parameter is *currently* unused, but could be useful for debugging. (It feels like it's a bit of a coincidence that at the moment we don't need to say which session is being detached.)

Reply:
Yes, I like leaving the parameter in but documenting we don't use it. Indeed, it's totally a detail implementaiton that we don't need to use the actual session we are detaching, to detach it.